### PR TITLE
jottacloud: implement SetModTime to support modtime-only changes

### PR DESCRIPTION
**Pull request by albertony**
#### What is the purpose of this change?

Implementing SetModTime function of Jottacloud backend, to add support for `rclone touch` on existing file, modtime-only changes of files in mounts etc.

Note: With the existing backend implementation's use of the allocate API, with its deduplication/resume features, whenever rclone wants to upload an identical file with changed timestamp, then it will be reduced to a "SetModTime" operation anyway: The allocate request updates the timestamp and returns that file is already COMPLETED, no need to upload data with a following update request. So in that sense, adding support for SetModTime does not change anything for regular sync/copy operations. What it does do is add support for modtime-only operations that does not resort to "re-upload".

Implementation note: Updates both Created and Modified timestamps, which also the existing Update function does, because Jottacloud seems to always update both anyway. If not including a Created value in the allocate request, Jottacloud will automatically set one (current time when server processes the request?).

#### Was the change discussed in an issue or in the forum before?

The following issues touches in on it, at least:

https://github.com/rclone/rclone/issues/5627
https://github.com/rclone/rclone/issues/1066

#### Checklist

- [X] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-pull-request).
- [ ] I have added tests for all changes in this PR if appropriate.
- [X] I have added documentation for the changes if appropriate.
- [X] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [X] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

File: backend/jottacloud/jottacloud.go [hunk 1]
```diff
@@ -1406,7 +1406,46 @@ func (o *Object) ModTime(ctx context.Context) time.Time {
...
+		Bytes:    o.size,
+		Created:  fileDate,
+		Modified: fileDate,
+		Md5:      o.md5,
+		Path:     path.Join(o.fs.opt.Mountpoint, o.fs.opt.Enc.FromStandardPath(path.Join(o.fs.root, o.remote))),
+	}
+
+	// send it
+	var response api.AllocateFileResponse
+	err = o.fs.pacer.CallNoRetry(func() (bool, error) {
```
Comment (ncw) [inline backend/jottacloud/jottacloud.go:1438]: Why is this `NoRetry`? We generally use that for uploads which read from an `io.Reader` which can't be rewound.

**Review by ncw — COMMENTED** [review]
Very nice!

Can you reference the relevant issues in your commit message please? If it completes them then `Fixes #1234` otherwise `#1234`.

Now for the important question...

Do the integration tests pass with your change?

File: backend/jottacloud/jottacloud.go (hunk 1 above)
Comment (albertony) [reply backend/jottacloud/jottacloud.go:1438]: No idea! I copied it from the existing upload implementation without thinking about this. I guess I should just do `Call`. Will fix.

Comment (albertony) [reply backend/jottacloud/jottacloud.go]: Done. Also removed the `response api.AllocateFileResponse` object, as it was not checked anyway, sending nil as argument saves a json parsing I guess.

**Comment by albertony** [comment]
> Do the integration tests pass with your change?

Yes. Ran through some fs tests as well, operations and sync. Get some errors (TestCopyFile, TestCopyFileCopyDest, TestSyncCopyDest), but identical results on master.

I think this should be good to go now.

File: backend/jottacloud/jottacloud.go (hunk 1 above)
Comment (albertony) [reply backend/jottacloud/jottacloud.go]: I think I will re-add the response, and instead actually check it. I managed to provoke an error that went unnoticed, although not very likely in real life cases, its probably better safe than sorry.

**Review by ncw — APPROVED** [review]
I think this looks good to go now.

Please merge when ready - thank you :-)

File: backend/jottacloud/jottacloud.go [hunk 2]
```diff
@@ -1406,7 +1406,45 @@ func (o *Object) ModTime(ctx context.Context) time.Time {
...
-	return fs.ErrorCantSetModTime
+	// make sure metadata is available, we need its current size and md5
+	err := o.readMetaData(ctx, false)
+	if err != nil {
+		fs.Logf(o, "Failed to read metadata: %v", err)
+		return err
+	}
+
+	// prepare allocate request with existing metadata but changed timestamps
+	o.fs.tokenRenewer.Start()
```
[the remark this answers is not in the document]
Comment (albertony) [reply backend/jottacloud/jottacloud.go]: Thanks for the info. I decided to remove it.

**Pull request merged**